yb-voyager/src/queryparser/traversal_plpgsql.go

```diff
@@ -0,0 +1,230 @@
+	PERFORM pg_advisory_lock(sender_id);
+	PERFORM pg_advisory_lock(receiver_id);
+
+	-- Deduct the amount from the sender's account
+	UPDATE accounts
+	SET balance = balance - transfer_amount
+	WHERE account_id = sender_id;
+
+	-- Add the amount to the receiver's account
+	UPDATE accounts
+	SET balance = balance + transfer_amount
+	WHERE account_id = receiver_id;
+
+	-- Release the advisory locks
+	PERFORM pg_advisory_unlock(sender_id);
+	PERFORM pg_advisory_unlock(receiver_id);
+
+END;
+$$;
+
+parsed json -
+
+	{
+	  "PLpgSQL_function": {
+	    "datums": [
+	      ...
+	    ],
+	    "action": {
+	      "PLpgSQL_stmt_block": {
+	        "lineno": 2,
+	        "body": [
+	          {
+	            "PLpgSQL_stmt_perform": {
+	              "lineno": 4,
+	              "expr": {
+	                "PLpgSQL_expr": {
+	                  "query": "SELECT pg_advisory_lock(sender_id)",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+				... similar to above
+	          },
+	          {
+	            "PLpgSQL_stmt_execsql": {
+	              "lineno": 8,
+	              "sqlstmt": {
+	                "PLpgSQL_expr": {
+	                  "query": "UPDATE accounts \n    SET balance = balance - transfer_amount \n    WHERE account_id = sender_id",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+				.... similar to above
+	          },
+	          {
+	            ... similar to below
+	          },
+	          {
+	            "PLpgSQL_stmt_perform": {
+	              "lineno": 19,
+	              "expr": {
+	                "PLpgSQL_expr": {
+	                  "query": "SELECT pg_advisory_unlock(receiver_id)",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+	            "PLpgSQL_stmt_return": {}
+	          }
+	        ]
+	      }
+	    }
+	  }
+	}
+*/
+func TraversePlPgSQLJson(fieldValue interface{}, plPgSqlStatements *[]string) {
+	fieldMap, isMap := fieldValue.(map[string]interface{})
+	fieldList, isList := fieldValue.([]interface{})
+	switch true {
+	case isMap:
+		for k, v := range fieldMap {
+			switch k {
+			// base case of recursive calls to reach this PLPGSQL_EXPR field in json which will have "query" field with statement
+			case PLPGSQL_EXPR:
+				expr, ok := v.(map[string]interface{})
+				if ok {
+					query, ok := expr[QUERY]
+					if ok {
+						q := formatExprQuery(query.(string)) // formating the query of parsed json if required
+
+						*plPgSqlStatements = append(*plPgSqlStatements, q)
+					}
+				}
+			default:
+				TraversePlPgSQLJson(v, plPgSqlStatements)
+			}
+		}
+	case isList:
+		//In case the value of a field is not a <key , val> but a list of <key, val> e.g. "body"
+		for _, l := range fieldList {
+			TraversePlPgSQLJson(l, plPgSqlStatements)
+		}
+	}
```

Comment:
consider using https://github.com/tidwall/gjson module as it simplifies the traversal of json strings

For example traversal plpgsql code can be converted to this::

```
func ExtractQueriesUsingGjson(parsedJson string) []string {
    var queries []string

    // Navigate to the action block
    actions := gjson.Get(parsedJson, "PLpgSQL_function.action")

    // Iterate over each statement in the body
    actions.Get("PLpgSQL_stmt_block.body").ForEach(func(key, value gjson.Result) bool {
        stmtType := key.String()

        switch stmtType {
        case "PLpgSQL_stmt_execsql":
            // Extract query from 'sqlstmt.PLpgSQL_expr.query'
            query := value.Get("sqlstmt.PLpgSQL_expr.query").String()
            formattedQuery := formatExprQuery(query)
            queries = append(queries, formattedQuery)

        case "PLpgSQL_stmt_perform":
            // Extract query from 'expr.PLpgSQL_expr.query'
            query := value.Get("expr.PLpgSQL_expr.query").String()
            formattedQuery := formatExprQuery(query)
            queries = append(queries, formattedQuery)

        // Add more cases for other statement types as needed
        // For example: PLpgSQL_stmt_if, PLpgSQL_stmt_loop, etc.

        default:
            // Handle or ignore other statement types
        }

        return true // Continue iteration
    })

    return queries
}
```